great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -792,6 +1042,17 @@ def _pandas_map_condition_rows(
         if filter_column_isnull:
             df = df[df[column_name].notnull()]
 
+    elif "column_list" in accessor_domain_kwargs:
+        column_list = accessor_domain_kwargs["column_list"]
+
+        for column_name in column_list:
+            if column_name not in metrics["table.columns"]:
+                raise ge_exceptions.ExecutionEngineError(
+                    message=f'Error: The column "{column_name}" in BatchData does not exist.'
+                )
+
+        df = df[column_list]
+
```

Comment:
This case is very similar to the one I pointed out in `_pandas_map_condition_index` above, except that here it is *necessary* to not do this filter, since our goal is to return entire rows. In fact, we should ensure a test exists that reveals this, because it would have been a bug.

```suggestion
```

Reply:
@jcampbell I believe that this check is needed for now, because it is analogous to the following:
```
    if "column" in accessor_domain_kwargs:
        column_name = accessor_domain_kwargs["column"]

        if column_name not in metrics["table.columns"]:
            raise ge_exceptions.ExecutionEngineError(
                message=f'Error: The column "{column_name}" in BatchData does not exist.'
            )
```
In other words, if, say, a `Checkpoint` configuration includes an expectation with a typo in a column name, we signal an error.  The reason we did this in the first place at the time of our work on `Checkpoints` was because without these checks, the error message would be completely absorbed by the backend (e.g., `Pandas`) with absolutely no feedback to the user.  This was "a bear" to debug back then.  With these checks, the user knows what went wrong.